### DISCUSSION TITLE
Add/Remove buttons on a collection

### DISCUSSION THREAD

**Discussion by lsmoker** [jrief/django-formset#123] [Q&A]
Is there a good way to remove these buttons when you only want to allow viewing or editing of existing records? For now, I'm removing the buttons with javascript, but that feels hackish.

**Comment by lsmoker** [comment]
Could this be done by overriding the `formset/default/collection.html` template and how would that be done?

**Comment by jrief** [comment]
I assume you mean a collection with siblings. Then the question is, why you want  to remove them? Is it because the user shall not be able to add and remove siblings because `min_siblings == max_siblings`?

Hmm, good point. I agree that in such a situation, those buttons indeed don't make sense. I think this then is a valid feature request. 

Is that your use-case?

**Comment by lsmoker** [comment]
The use case is that the records/siblings will only be added or removed by another process, and the form is only being used to present the information and allow updates to one field.

**Comment by lsmoker** [comment]
Overriding formset/default/collection.html and subclassing FormRenderer like so:

```
from django.template.loader import get_template

class NoAddRemoveFormRenderer(FormRenderer):
    def get_template(self, template_name):
        return get_template(template_name)

class LoanCollection(FormCollection):
    default_renderer = NoAddRemoveFormRenderer()
    template_name = 'formset/default/collection.html'
    min_siblings = 0
    ...
```
... "works" to remove the buttons but then django-formset.js throws an error - `<django-form-collection> with siblings requires child element <button class="remove-collection">`

So back to my js hack.

**Comment by lsmoker** [comment]
Actually, CSS makes it simple...

```
<style>
django-formset django-form-collection button.add-collection,button.remove-collection {
    display: None;
}
</style>
```

Not pythonic at all though...
